### PR TITLE
Add 'conda env list --size'

### PR DESCRIPTION
**Pull request by nok**
Add argument `--size` to command `conda env list` to compute the disk usage of each environment. It's useful when the user has many environments and want to clean up old and large environments.

And here you can see the result:

```bash
conda env list --size
```

```bash
# conda environments:
#
base                 * /Users/darius/repos/conda/devenv 470.09 MB
env_a                  /opt/miniconda/envs/env_a        117.17 MB
env_b                  /opt/miniconda/envs/env_b        739.96 MB
```

It's similar to the command `find ./envs/prefix -type f -exec ls -l {} \; | awk '{sum += $5} END {print sum}'` which is more accurate than `du -sh ./envs/`.

By the way the computation can be slow, because of that it's an optional argument.

Thanks, Darius

### CODE REVIEW

**Comment by msarahan** [comment]
This is a very good idea.  I'm re-running the tests, as I don't understand if the failures were related to your code changes or not.  Have you signed the CLA yet?  If not, please do so.  Unfortunately, we can't accept contributions from people who haven't signed the CLA.

**Comment by nok** [comment]
I finished the CLA request. Should I send the response to kfranz@anaconda.com?

**Comment by msarahan** [comment]
I got the response. I'll process it now. It's a manual process. It seems like the tests really are failing due to something in this process, though, and we need to look at why. I'll try to find some time.

File: conda/cli/common.py [hunk 1]
```diff
@@ -177,29 +179,57 @@ def stdout_json_success(success=True, **kwargs):
...
+    def compute_size(prefix):
+        total_size = 0
+        for dirpath, dirnames, filenames in walk(prefix):
+            for filename in filenames:
+                path = join(dirpath, filename)
+                if not islink(path):
+                    total_size += getsize(path)
+        return total_size
+
+    def convert_size(size_bytes):
```
[the remark this answers is not in the document]
Comment (nok) [reply conda/cli/common.py]: Sure, thanks for the hint for using `human_bytes`.

File: conda/cli/common.py [hunk 2]
```diff
@@ -177,29 +179,57 @@ def stdout_json_success(success=True, **kwargs):
...
-        print("#")
-
-    def disp_env(prefix):
-        fmt = '%-20s  %s  %s'
-        default = '*' if prefix == context.default_prefix else ' '
+def print_envs_list(known_conda_prefixes, with_size=False):
+    print("# conda environments:")
+    print("#")
+
+    def compute_size(prefix):
```
[the remark this answers is not in the document]
Comment (nok) [reply conda/cli/common.py]: This is a good suggestion. What I did so far:

```python
def comp_cached_size(prefix):
    total_size = 0
    for rcrd in PrefixData(prefix).iter_records():
        if 'size' in rcrd:
            total_size += int(rcrd.size)
    return total_size

# size = sum([int(rcrd.size) for rcrd in PrefixData(prefix).iter_records() if 'size' in rcrd])

size = human_bytes(comp_cached_size(prefix)) if with_size else ''
```

But the computed sums are tiny. What's the exact definition of `size` in the `/envs/<env_name>/conda-meta/<package_name>.json` meta files? Amount of files? Or maybe the sum of compressed file sizes? Sum of unresolves file sizes?

**Comment by travishathaway** [comment]
This pull request has become stale and is therefore being closed. Please re-open if you would like to continue development on this.

**Pull request closed without merging**